test/integration/consul-container/libs/service/assets/Dockerfile-consul-envoy

```diff
@@ -0,0 +1,7 @@
+# Note this arg has to be before the first FROM
+ARG ENVOY_VERSION
+
+FROM consul:local as consul
+
+FROM docker.mirror.hashicorp.services/envoyproxy/envoy:v${ENVOY_VERSION}
+COPY --from=consul /bin/consul /bin/consul
```

Comment:
How this get into the CI?

Reply:
The file or the base image? 
* The base image happens in [prior step in the pipeline](https://github.com/hashicorp/consul/blob/4c359dcb696dfbdf234d2b68c756cf1dd137d65b/.circleci/config.yml#L840-L848).
* The file is passed into testcontainers [docker context via go:embed](https://github.com/hashicorp/consul/pull/15084/files#diff-a20cab050471215df956d350bb1352623fa422052f64add57fb0bee080e27818R17-R18), then built dynamically when the test runs. This seemed similar enough to how the bats tests work.

Reply:
oh I missed the go:embed directive 😓